celery_progress/static/celery_progress/celery_progress.js

```diff
@@ -79,7 +86,13 @@ class CeleryProgressBar {
             if (data.success === true) {
                 this.onSuccess(this.progressBarElement, this.progressBarMessageElement, this.getMessageDetails(data.result));
             } else if (data.success === false) {
-                this.onTaskError(this.progressBarElement, this.progressBarMessageElement, this.getMessageDetails(data.result));
+                if (data.state === 'RETRY') {
+                    this.onRetry(this.progressBarElement, this.progressBarMessageElement, data.result.message, data.result.when);
+                    done = false;
+                    delete data.result;
```

Comment:
I understand why this is needed. The existing code is confusing and should probably be changed in the future.

Reply:
I agree, while the previous code works with few states, it does not make extending easy. I originally intended to add more than just RETRY support in this PR, but trying to support REVOKED and IGNORED in this state was a bit hairy.